Jenkinsfile

```
@@ -30,6 +30,34 @@ boolean doc_only_change() {
     return rc == 1
 }
 
+boolean release_candidate() {
+    if (!sh(label: "Determine if building (a PR of) an RC",
+          script: "git diff-index --name-only HEAD^ | grep -q TAG && " +
+                  "grep -i '[0-9]rc[0-9]' TAG",
+          returnStatus: true)) {
+        return true
+    }
+    return false
+}
+
+def scons_faults_enabled() {
+    // if the faults_enabled pragma is false or it a release candidate; disable fault injection
+    if ((cachedCommitPragma(pragma: 'faults-enabled', def_val: 'true') != 'true') || release_candidate()) {
+        return "BUILD_TYPE=release"
+    } else {
+        return "BUILD_TYPE=dev"
+    }
+}
+
+def rpm_faults_enabled() {
```

This function seems like a duplicate of `scons_faults_enabled()` but with different return strings.  It could be DRYed out by taking an argument of which kind of string it's to return.  Or it could be made boolean and a simplified `scons_faults_enabled()` and `rpm_faults_enabled()` to just call the common boolean returning function.

Also:

```suggestion
def rpm_faults_args() {
```